### PR TITLE
fix: date Picker rehaul

### PR DESCRIPTION
**Pull request by harshilp24**

### CODE REVIEW

File: website/docs/reference/widgets/datepicker.md [hunk 1]
```diff
@@ -1,132 +1,168 @@
...
-Datepicker is used to capture the date and time input by a user. It can be used to filter data based on the input date range as well as to capture personal information such as date of birth.
+This document explains how to use a Datepicker widget to capture date and time input from users.
 
-<VideoEmbed host="youtube" videoId="MFflGf3K324" title="How to use Datepicker Widget" caption="How to use Datepicker Widget"/>
+<VideoEmbed host="youtube" videoId="MFflGf3K324" title="Configure Datepicker Widget" caption="Configure Datepicker Widget"/>
 
-## Properties
 
-Properties allow you to edit the widget, connect it with other widgets and customize the user actions.
+## Set default date
```
Comment (danciaclara) [inline website/docs/reference/widgets/datepicker.md:8]:
```suggestion
## Set date and time
```

File: website/docs/reference/widgets/datepicker.md [hunk 2]
```diff
@@ -1,132 +1,168 @@
...
+When working with a Datepicker widget, you may want to set a default date value to be displayed to the user before they select anything. This can be useful for providing context or a starting point for the user.
```
Comment (danciaclara) [inline website/docs/reference/widgets/datepicker.md:9]:
```suggestion
When working with the Datepicker widget, you may want to set a default date value to be displayed to the user.
```

File: website/docs/reference/widgets/datepicker.md [hunk 3]
```diff
@@ -1,132 +1,168 @@
...
 
-### Widget Properties
+To set a default date, you can use the **Default Date** property. You can set this property to any valid date format that the widget supports.
```
Comment (danciaclara) [inline website/docs/reference/widgets/datepicker.md:11]:
```suggestion
To set the date and time, you can use the **Default Date** property. You can set this property to any valid date format that the widget supports. 
```

File: website/docs/reference/widgets/datepicker.md [hunk 4]
```diff
@@ -1,132 +1,168 @@
...
-Properties allow you to edit the widget, connect it with other widgets and customize the user actions.
+## Set default date
+When working with a Datepicker widget, you may want to set a default date value to be displayed to the user before they select anything. This can be useful for providing context or a starting point for the user.
 
-### Widget Properties
+To set a default date, you can use the **Default Date** property. You can set this property to any valid date format that the widget supports. 
 
-These properties allow you to edit the Datepicker widget. All of these properties are present in the property pane of the widget. The following table lists all the widget properties.
+### Date format
+Appsmith provides several date formats that you can choose from, such as:
```
Comment (danciaclara) [inline website/docs/reference/widgets/datepicker.md:14]:
```suggestion
Appsmith provides several date formats that you can choose from in the **Date Format** property.
```

File: website/docs/reference/widgets/datepicker.md [hunk 5]
```diff
@@ -1,132 +1,168 @@
...
-| **First Day Of Week**  | Sets which day of the week appears first within the calendar of the Datepicker's menu. Accepts number values between 0 and 6.<br/>0 represents Sunday, 1 represents Monday, and so on.                                                                                                                                                                                                                                                                                                               |
-| [**Height**](/reference/widgets/#height)        | It configures how a widget’s height reacts to content changes. It has three possible configurations:<br/>**Fixed**: The height of the widget remains as set using drag and resize.<br/>**Auto Height**: The height of the widget reacts to content changes.<br/>  **Auto Height with limits**: Same as Auto height, with a configurable option to set the minimum and maximum number of rows that can be occupied by the widget.                                      |
+* ISO 8601
```
Comment (danciaclara) [inline website/docs/reference/widgets/datepicker.md:16]: no need to list the date formats

File: website/docs/reference/widgets/datepicker.md [hunk 6]
```diff
@@ -1,132 +1,168 @@
...
+* YYYY-MM-DD
+* MM-DD-YYYY
+* DD-MM-YYYY
+* MM/DD/YYYY
+* DD/MM/YYYY
+* DD/MM/YY
 
 
-### Binding Properties
+If you need more advanced date formatting options, you can use the [**Moment.js**](https://momentjs.com/docs/) library which is installed by default in Appsmith. Moment.js is a JavaScript library for parsing, validating, manipulating, and formatting dates. With Moment.js, you can define your own custom date formats using a simple syntax.
```
Comment (danciaclara) [inline website/docs/reference/widgets/datepicker.md:33]:
```suggestion
If you need more advanced date formatting options, you can use the built-in [**Moment.js**](https://momentjs.com/docs/) library in Appsmith. With Moment.js, you can define your own custom date formats using JavaScript.
```

File: website/docs/reference/widgets/datepicker.md [hunk 7]
```diff
@@ -1,132 +1,168 @@
...
 
-These properties allow you to bind your Datepicker widget with any other widget in queries or JS objects. The following table lists all the binding properties.
+To use Moment.js with Appsmith, you can simply write JavaScript code in the widget's **Date Format** property. For example, you can use the following code to set the default date value to **today's date** in a specific format:
```
Comment (danciaclara) [inline website/docs/reference/widgets/datepicker.md:35]:
```suggestion
To use Moment.js with Appsmith, you can write JavaScript code in the widget's **Date Format** property using the mustache syntax `{{}}`. For example, you can use the following code to set the default date value to **today's date** in a specific format:
```

File: website/docs/reference/widgets/datepicker.md [hunk 8]
```diff
@@ -1,132 +1,168 @@
...
 
-| Binding Property  | Description                                                                                                                                                    |
-| ----------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| **formattedDate** | Contains the date value currently selected within the Datepicker widget. This value changes if the default value is updated or if the user inputs a new value. |
-| **selectedDate**  | This is the ISO date string selected in the Datepicker widget. This value changes if the default value is updated or if the user inputs a new value.           |
-| **isDisabled**    | Reflects the state of the widget's **Disabled** setting _(bool)_.                                                                                              |
-| **isVisible**     | Reflects the state of the widget's **Visible** setting _(bool)_.                                                                                               |
+```JS
+{{moment().format('YYYY-MM-DD')}}
```
Comment (danciaclara) [inline website/docs/reference/widgets/datepicker.md:38]: In the Date format field I think you only need to specify the format in single quotes inside mustache syntax. Check this with the App viewers pod

File: website/docs/reference/widgets/datepicker.md [hunk 9]
```diff
@@ -1,132 +1,168 @@
...
-| ------------------ | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| **onDateSelected** | Sets an action to take place when a date is selected. Can be set from the GUI list of common actions ([examples here](../appsmith-framework/widget-actions/)), or you can define a custom JavaScript function to call instead. |
-| **onFocus** | Sets an action to take place when a datepicker is focused. Can be set from the GUI list of common actions ([examples here](../appsmith-framework/widget-actions/)), or you can define a custom JavaScript function to call instead. |
-| **onBlur** | Sets an action to take place when a datepicker loses focus. Can be set from the GUI list of common actions ([examples here](../appsmith-framework/widget-actions/)), or you can define a custom JavaScript function to call instead. |
+## Display date dynamically
 
-### Label
+You can display the date dynamically based on user interaction or by fetching the date from a query response.
```
Comment (danciaclara) [inline website/docs/reference/widgets/datepicker.md:46]: mention that this also provided in the Default date property

File: website/docs/reference/widgets/datepicker.md [hunk 10]
```diff
@@ -1,132 +1,168 @@
...
-| **Width**     | Sets the width of the label. The number represents how many characters/columns wide the label is.    |   |
+---
 
-| Label Styles         | Description                                                                                                                          |   |
-| -------------------- | ------------------------------------------------------------------------------------------------------------------------------------ | - |
-| **Text Color**       | Sets the text color for the label. Accepts valid CSS [`color` ](https://developer.mozilla.org/en-US/docs/Web/CSS/color)values.       |   |
-| **Text Size**        | Sets the size of the label font. Accepts valid CSS [`font-size`](https://developer.mozilla.org/en-US/docs/Web/CSS/font-size) values. |   |
-| **Label Font Style** | Toggles font styles (**bold** or _italic)._                                                                                          |   |
+**Example**: suppose you want to display the date of birth of users in the [mock database](https://docs.appsmith.com/core-concepts/connecting-to-data-sources/connecting-to-databases#sample-databases).
```
Comment (danciaclara) [inline website/docs/reference/widgets/datepicker.md:51]: write similar to the one on JSON form where we help the user understand the context better

File: website/docs/reference/widgets/datepicker.md [hunk 11]
```diff
@@ -1,132 +1,168 @@
...
+```
+Then, set the `onDateSelected` property of the Datepicker widget to run the`updateDob` query. 
 
-| Style Property    | Description                                                                                                                                                                      |
-| ----------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| **Border Radius** | Rounds the corners of the widget's outer edge. With JS enabled, this accepts valid CSS [`border-radius`](https://developer.mozilla.org/en-US/docs/Web/CSS/border-radius) values. |
-| **Box Shadow**    | Casts a drop shadow from the frame of the widget. With JS enabled, this accepts valid CSS [`box-shadow`](https://developer.mozilla.org/en-US/docs/Web/CSS/box-shadow) values.    |
 
-### Month Picker
+### Retrieve data for a range
```
Comment (danciaclara) [inline website/docs/reference/widgets/datepicker.md:84]:
```suggestion
### Filter data for a date range
```

File: website/docs/reference/widgets/datepicker.md [hunk 12]
```diff
@@ -1,132 +1,168 @@
...
+---
+**Example**: lets take the same users example, if you want to update the date of birth, you can create a new query called `updateDob` with an SQL statement like:
 
-Style properties allow you to change the look and feel of the widget.
+```sql
+UPDATE users
+  SET dob = '{{DatePicker.formattedDate}}'
+  WHERE id = {{tblUserData.selectedRow.id}};
+```
+Then, set the `onDateSelected` property of the Datepicker widget to run the`updateDob` query.
```
Comment (danciaclara) [inline website/docs/reference/widgets/datepicker.md:81]:
```suggestion
Then, set the `onDateSelected` event listener of the Datepicker widget to run the`updateDob` query. 
```

File: website/docs/reference/widgets/datepicker.md [hunk 13]
```diff
@@ -1,132 +1,168 @@
...
 
-### Month Picker
+### Retrieve data for a range
 
-To select just a month rather than a particular date, you can convert the [**Select widget**](./select.md) into a month picker. Drag a select widget to the canvas and enter the following snippet:
+Retrieving data for a specific date involves using a query to retrieve data that falls within a specified date range.
 
-```
-{{moment.months().map((v) => { return {label: v, value: v.toLowerCase()}})}}
+For example, suppose you want to allows users to filter data for specific dates, such as retrieving data of users born between `01/01/1980` and `01/01/201`. To accomplish this, you can add two date pickers to your canvas. Then, create a new query called `filterdata` with the SQL statement:
```
Comment (danciaclara) [inline website/docs/reference/widgets/datepicker.md:88]: use the same example format

File: website/docs/reference/widgets/datepicker.md [hunk 14]
```diff
@@ -1,132 +1,168 @@
...
+
+Properties allow you to edit the widget, connect it with other widgets and customize the user actions.
+
+
+### Widget properties
+
+These properties are present in the property pane of the widget. The following table lists all the widget properties.
+
+
+|  Property   | Data type |  Description                                                                                                                                                                      |
```
Comment (danciaclara) [inline website/docs/reference/widgets/datepicker.md:112]: update data types

**Review by danciaclara — CHANGES_REQUESTED** [review]

File: website/docs/reference/widgets/datepicker.md [hunk 15]
```diff
@@ -1,132 +1,172 @@
...
 
-| Binding Property  | Description                                                                                                                                                    |
-| ----------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| **formattedDate** | Contains the date value currently selected within the Datepicker widget. This value changes if the default value is updated or if the user inputs a new value. |
-| **selectedDate**  | This is the ISO date string selected in the Datepicker widget. This value changes if the default value is updated or if the user inputs a new value.           |
-| **isDisabled**    | Reflects the state of the widget's **Disabled** setting _(bool)_.                                                                                              |
-| **isVisible**     | Reflects the state of the widget's **Visible** setting _(bool)_.                                                                                               |
 
-### Events
+### Set date in a timezone
```
Comment (danciaclara) [inline website/docs/reference/widgets/datepicker.md:24]:
```suggestion
```

File: website/docs/reference/widgets/datepicker.md [hunk 16]
```diff
@@ -1,132 +1,172 @@
...
 
-You can define functions that will be called when these events are triggered in the widget.
+When working with dates and time, you may want to display them in a specific time zone. You can achieve this using Moment.js and the `moment-timezone` library.
```
Comment (danciaclara) [inline website/docs/reference/widgets/datepicker.md:26]:
```suggestion
```

File: website/docs/reference/widgets/datepicker.md [hunk 17]
```diff
@@ -1,132 +1,172 @@
...
 
-### Label
+```js
+{{
+  moment(DatePicker.selectedDate).tz("Asia/Kolkata").format()
+}}
+```
 
-The property hosts a group of configurations that you can use to associate a display name and define a placement for the widget. These properties are usually useful when you want to design forms that follow a defined alignment for your form fields and give a professional look to your forms. Below are the properties that you can use:
+This code creates a `Moment.js` object from the `DatePicker.selectedDate` property, sets the time zone to `"Asia/Kolkata"` using the `tz()` function, and formats the resulting date and time using the default format. To learn more, refer to [MomentJS timezone documentation](https://momentjs.com/timezone/docs/).
```
Comment (danciaclara) [inline website/docs/reference/widgets/datepicker.md:37]:
```suggestion
```

File: website/docs/reference/widgets/datepicker.md [hunk 18]
```diff
@@ -1,132 +1,172 @@
...
-| **Alignment** | Sets whether the label is left- or right-aligned.                                                    |   |
-| **Width**     | Sets the width of the label. The number represents how many characters/columns wide the label is.    |   |
 
-| Label Styles         | Description                                                                                                                          |   |
-| -------------------- | ------------------------------------------------------------------------------------------------------------------------------------ | - |
-| **Text Color**       | Sets the text color for the label. Accepts valid CSS [`color` ](https://developer.mozilla.org/en-US/docs/Web/CSS/color)values.       |   |
-| **Text Size**        | Sets the size of the label font. Accepts valid CSS [`font-size`](https://developer.mozilla.org/en-US/docs/Web/CSS/font-size) values. |   |
-| **Label Font Style** | Toggles font styles (**bold** or _italic)._                                                                                          |   |
+### Display date dynamically
```
[the remark this answers is not in the document]
Comment (danciaclara) [reply website/docs/reference/widgets/datepicker.md:40]: should be after line 11 example

File: website/docs/reference/widgets/datepicker.md [hunk 19]
```diff
@@ -1,132 +1,172 @@
...
+These are functions that are called when event listeners are triggered in the widget. Use [actions](/reference/appsmith-framework/widget-actions) to execute tasks based on user events.
+
+
+| Event              | Description                                                                                                                                                                                                                       |
+| ------------------ | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **onDateSelected** | Triggers an action when a user selects a date.  |
+| **onFocus** | Triggers an action when a Datepicker widget is focused. |
+| **onBlur** | Triggers an action when a Datepicker widget loses focus. |
+
+## Sample apps
```
Comment (danciaclara) [inline website/docs/reference/widgets/datepicker.md:164]: can we keep this above properties?

File: website/docs/reference/widgets/filepicker.md [hunk 20]
```diff
@@ -1,6 +1,6 @@
 # Filepicker
 
-This document guides you on how to use the Filepicker widget for selecting and uploading various file types, including images, videos, audio, text files, and more.
+This document explains how to use a datapicker widget to capture date and time input from users.
```
Comment (danciaclara) [inline website/docs/reference/widgets/filepicker.md:3]: why is this here?

**Review by danciaclara — CHANGES_REQUESTED** [review]

File: website/docs/reference/widgets/filepicker.md (hunk 20 above)
Comment (harshilp24) [reply website/docs/reference/widgets/filepicker.md:3]: Wait a min, I have no idea :(  I'll fix this

**Review by danciaclara — APPROVED** [review]
